### PR TITLE
[FLINK-32821][examples] Include flink-connector-datagen for streaming examples

### PR DESCRIPTION
**Pull request by X-czh**
## What is the purpose of the change

Fix the issue that 4 out of 7 (MatrixVectorMul & Iteration & SessionWindowing & TopSpeedWindowing) Flink streaming examples cannot run because datagen connector is not packaged into the jars.

## Brief change log

Package datagen connector into the self-contained example jars using Maven shade plugin.

## Verifying this change

- Verified that these examples in a local standalone cluster.
- Verified that the final jar does not contain any other classes besides connector-datagen and the target test classes.
- Added integrated tests to verify if all examples can be submitted successfully and verified that the tests failed before the fix and succeeded after the fix.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no
## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by X-czh** [comment]
@flinkbot run azure

**Review by huwh — COMMENTED** [review]
Thanks for preparing this PR. I left a comment.

And, I think we shouldn't package flink-examples-streaming and flink-connector-datagen into a uber-jar.
In my opinion, there are other modules that depend on flink-examples-streaming and we won't use  flink-examples-streaming jar as an example jar. 
Instead, the jar used as an example should be shaded to be a self-contained uber-jar, such as TopSpeedWindowing,MatrixVectorMul...

**Comment by X-czh** [comment]
Thanks @huwh for the comments. I've updated a new commit that did the following:
1. No longer package `flink-examples-streaming` and `flink-connector-datagen` into a uber-jar, as there are other modules that depend on `flink-examples-streaming` and we don't directly use `flink-examples-streaming` jar as an example jar.
2. Disable reduced dependency pom generation for `flink-example-streaming` as this will reduce dependencies (e.g. `flink-connector-datagen`) included in the shaded jar but required by other modules depending on `flink-examples-streaming` (e.g. `flink-examples-streaming-state-machine`).
3. Fix the wrong filtering setting in the previous commit that mistakenly excludes classes from `flink-connector-kafka`.
4. Remove the Scala version suffix when including `flink-examples-streaming` and `flink-connector-kafka` as they are Scala-free now.

Please review again when you are free, much thanks!

File: flink-examples/flink-examples-build-helper/flink-examples-streaming-state-machine/pom.xml [hunk 1]
```diff
@@ -71,15 +71,16 @@ under the License.
...
+									<include>org.apache.flink:flink-connector-kafka</include>
+									<include>org.apache.flink:flink-examples-streaming</include>
 									<include>org.apache.kafka:*</include>
 								</includes>
 							</artifactSet>
 
 							<filters>
 								<filter>
-									<artifact>org.apache.flink:flink-examples-streaming_*</artifact>
+									<artifact>org.apache.flink:flink-examples-streaming</artifact>
```
Comment (huwh) [inline flink-examples/flink-examples-build-helper/flink-examples-streaming-state-machine/pom.xml:83]: It's better to split these changes (fix wrong filtering and remove Scala version) into a hotfix commit at the top of this PR.

**Review by huwh — COMMENTED** [review]
Thanks for the update. Generally, we will separate the small fix changes as a independent commit and put it at the top of the PR. It could named as "[hotfix]xxxx"

File: flink-examples/flink-examples-build-helper/flink-examples-streaming-state-machine/pom.xml (hunk 1 above)
Comment (X-czh) [reply flink-examples/flink-examples-build-helper/flink-examples-streaming-state-machine/pom.xml:83]: Thanks for the comment, I'll separate the commits.

Comment (huwh) [reply flink-examples/flink-examples-build-helper/flink-examples-streaming-state-machine/pom.xml:83]: Usually, we add hotfix before other changes.

Comment (X-czh) [reply flink-examples/flink-examples-build-helper/flink-examples-streaming-state-machine/pom.xml:83]: Sure, I'll rearrange the commits order and force-push again. Thanks for pointing that.

**Review by huwh — APPROVED** [review]
LGTM. Verified TopSpeedWindowing/SessionWindowing/Iteration/StateMachineExample in a standalone cluster. 

It will be merged after CI passes.

**Review by huwh — APPROVED** [review]

**Pull request merged**